crates/matrix-sdk/src/room/timeline/mod.rs

```diff
@@ -386,6 +387,44 @@ impl Timeline {
         let response = room.send(content, Some(&txn_id)).await;
         self.inner.handle_local_event_send_response(&txn_id, response)
     }
+
+    /// Fetch unavailable details about the event with the given ID.
+    ///
+    /// This method tries to make all the requests it can. If an error is
+    /// encountered for a given request, it is forwarded with the
+    /// [`TimelineDetails::Error`] variant.
+    ///
+    /// # Arguments
+    ///
+    /// * `txn_id` - The transaction ID of the event to fetch details for, if
+    ///   any.
+    ///
+    /// * `event_id` - The event ID of the event to fetch details for, if any.
```

Comment:
I think we should probably introduce an enum like `TimelineKey` again for this (that borrows the underlying ID this time I guess). Or is there a reason a user would want to pass both transaction and event ID?

Reply:
I don't think there's a good reason to pass both.

I've actually been wondering if we should support local events, because the event will be overwritten when we receive it from the server and the details will be lost.

Also it's unlikely someone will send a reply to an event that is not in the timeline (otherwise where would they get the event ID?), so we could add the `RepliedToEvent` when creating the local echo.

Reply:
Okay so let's update this to just take an event ID then? We can always adjust later but that seems to make more sense now.